Packs/AWS-SecurityHub/Integrations/AWSSecurityHubEventCollector/AWSSecurityHubEventCollector.py

```diff
@@ -0,0 +1,251 @@
+import datetime
+
+import demistomock as demisto
+from AWSApiModule import *
+
+import urllib3
+from datetime import datetime
+
+import boto3
+
+# Disable insecure warnings
+urllib3.disable_warnings()
+
+VENDOR = 'AWS'
+PRODUCT = 'Security Hub'
+DATETIME_FORMAT = "%Y-%m-%dT%H:%M:%S.%fZ"
+DEFAULT_MAX_RESULTS = 1000  # Default maximum number of results to fetch
+
+
+def get_events(client: boto3.client, start_time: datetime | None = None,
+               end_time: datetime | None = None, id_ignore_list: list[str] | None = None, limit: int = 0) -> list[dict]:
+    """
+    Fetch events from AWS Security Hub.
+
+    Args:
+        client (boto3.client): Boto3 client to use.
+        start_time (datetime | None, optional): Start time to fetch events from. Required if end_time is set.
+        end_time (datetime | None, optional): Time to fetch events until. Defaults to current time.
+        limit (int): Maximum number of events to fetch. Defaults to 0 (no limit).
+        id_ignore_list (list[str] | None, optional): List of finding IDs to not include in the results.
+            Defaults to None.
+
+    Returns:
+        tuple[list, CommandResults]: A tuple containing the events and the CommandResults object.
+    """
+    kwargs = {'SortCriteria': [{'Field': 'UpdatedAt', 'SortOrder': 'asc'}]}
+    filters = {}
+
+    if end_time and not start_time:
+        raise ValueError('start_time must be set if end_time is used.')
+
+    if start_time:
+        filters['UpdatedAt'] = [{
+            'Start':
+                start_time.strftime(DATETIME_FORMAT),
+            'End':
+                end_time.strftime(DATETIME_FORMAT) if end_time else datetime.now().strftime('%Y-%m-%dT%H:%M:%SZ')
+        }]
+
+    if id_ignore_list:
+        ignore_filters = []
+        for event_id in id_ignore_list:
+            ignore_filters.append({'Value': event_id, 'Comparison': 'NOT_EQUALS'})
+
+        filters['Id'] = ignore_filters
+
+    if filters:
+        # We send kwargs because passing Filters=None to get_findings() tries to use a None value for filters,
+        # which raises an error.
+        kwargs['Filters'] = filters
+
+    events = []
+
+    while True:
+        # The API only allows a maximum of 100 results per request. Using more raises an error.
+        if limit and limit - len(events) < 100:
+            kwargs['MaxResults'] = limit - len(events)
+
+        else:
+            kwargs['MaxResults'] = 100
```

Comment:
Is there some API rate limit?
If there is, maybe we need to limit this function until 100 results and not to loop